src/integrator/sycl/collocation_device.cpp

```diff
@@ -18,15 +18,23 @@ namespace sycl       {
                                  const T *pts_device, T *eval_device,
                                  cl::sycl::queue *queue) {
 
-        cl::sycl::range<2> threads(16, 16);
-        cl::sycl::range<2> blocks(util::div_ceil(npts, threads[0]),
-                                  util::div_ceil(nshells, threads[1]) );
+        cl::sycl::range<3> threads(16, 16, 1);
+        cl::sycl::range<3> blocks(util::div_ceil(npts, threads[0]),
+                                  util::div_ceil(nshells, threads[1]),
+                                  1);
 
         GAUXC_SYCL_ERROR( queue->submit([&](cl::sycl::handler &cgh) {
                 auto global_range = blocks * threads;
 
-                cgh.parallel_for( cl::sycl::nd_range<2>(global_range, threads),
-                    [=](cl::sycl::nd_item<2> item_ct) {
+                cgh.parallel_for(
+                    cl::sycl::nd_range<3>(cl::sycl::range<3>(global_range.get(2),
```

Comment:
Why not just construct the `blocks` / `threads` ranges in the correct order to begin with?

Also, didn't we decide that `nd_range` was overkill here (since there's no reductions)? Would it be better to just use a regular range here and let SYCL determine how it wants to schedule it?

Reply:
done. `range` and `nd_range` have been appropriately placed where needed.